_posts/2023-04-19_esextractor.md

```diff
@@ -0,0 +1,95 @@
+# ESExtractor: how to integrate a dependency free library to CTS
+
+Since the [Vulkan CTS](https://github.com/KhronosGroup/VK-GL-CTS) is now able to test and check [Vulkan Video support](https://www.khronos.org/news/press/vulkan-sdk-is-vulkan-video-ready)
+ including video decoding, it was necessary to define the kind of media container to be used inside the test cases and the library
+to extract the necessary encoded data.
+
+In a first attempt, the [FFMpeg media toolkit](https://ffmpeg.org/) has been chosen to extract the video packets from the A/V ISO base media
+format chosen as a container reference. This library was provided as a binary package and loaded dynamically at each
+test run.
+
+As Vulkan video aims to test only *video* contents, it was not necessary to choose a complex media container,
+so first all the videos have been converted to the elementary stream format for
+h264 and H265 contents.
+This is a very elementary format based on MPEG start code and NAL unit identification.
+
+As FFMpeg is quite heavy, monolithic package and might cause issues to support multiple platform and architecture,
+a first attempt to replace it, was to use GStreamer and its helper library, [demuxeres](https://github.com/Igalia/GstVkVideoParser/tree/main/lib/demuxeres)
+It was a smaller package but need to back binary packaged to avoid the glib/gstreamer dependencies.
+For the CTS community it was a no go because of binary package which could lead to issues on niche platform supported by CTS.
+
+So at [Igalia](https://www.igalia.com/), we decided to implement a custom library, out of any dependencies, written in c++ to be compliant with
```

Comment:
to implement a dependency-free, custom library, written in C++ to be compliant with the CTS and simple to integrate into any build system